src/core/message.ts

```diff
@@ -288,4 +298,20 @@ export class DefaultMessage implements Message {
   equal(message: Message): boolean {
     return this.serial === message.serial;
   }
+
+  with(event: MessageEventPayload): Message {
+    if (event.type === MessageEvents.Created) {
+      throw new ErrorInfo('Cannot apply a created event to a message', 40000, 400);
```

Comment:
We use lowercase for our messages elsewhere, so should follow the same pattern here

Reply:
Also - do we just want to make passing in `MessageEvents.Created` a no-op, rather than an error?

Reply:
That was my initial attempt https://github.com/ably/ably-chat-js/pull/445#discussion_r1907068897

I think error because if users end up calling with(event) for creates, it suggests they haven't processed the create separately. The error will give a clear signal something's wrong (better DX when compared to new messages not showing and having no errors).